app/src/main/java/io/ably/example/androidpushexample/MainActivity.kt

```diff
@@ -248,9 +249,23 @@ class MainActivity : AppCompatActivity() {
 		return true
 	}
 
-    fun clientId(): String{
-        return Settings.Secure.getString(getContentResolver(), Settings.Secure.ANDROID_ID)
-    }
+	fun clientId():String {
+		var pref = PreferenceManager.getDefaultSharedPreferences(this)
+		if (pref.contains("clientId")) {
+			return pref.getString("clientId", Settings.Secure.getString(getContentResolver(), Settings.Secure.ANDROID_ID))
```

Comment:
Will this default `ANDROID_ID` value ever get used?

Reply:
I prefer returning some value instead of "null". So I thought of using ANDROID_ID.